### PR TITLE
Add instructions for installing Kyma on Gardener shoot cluster

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Added instructions for installing Kyma on Gardener-managed cluster (GCP/MS Azure)
- Updated/unified prerequisites across all installation instructions
- Updated/rephrased instructions for GCP Marketplace installation

**Related issue(s)**
#5348

### CODE REVIEW

File: docs/kyma/04-03-cluster-installation.md [hunk 1]
```diff
@@ -331,5 +318,61 @@ For MacOS, run:
     ```
 
 
+  </details>
+  <details>
+  <summary>
+  Gardener
+  </summary>
+
+Install Kyma on a [GKE](https://cloud.google.com/kubernetes-engine/) or [AKS](https://azure.microsoft.com/services/kubernetes-service/) cluster deployed through [Gardener](https://gardener.cloud/).
```
Comment (pbochynski) [inline docs/kyma/04-03-cluster-installation.md]: GKE and AKS are Kubernetes services. Gardener deploys K8S directly on VMs provided by GCP or Azure. Please change the links.

File: docs/kyma/04-03-cluster-installation.md [hunk 2]
```diff
@@ -331,5 +318,61 @@ For MacOS, run:
...
+
+1. Go to [this](https://github.com/kyma-project/kyma/releases/) page and choose the release you want to install.
+
+2. Export the release version as an environment variable. Run:
+
+    ```
+    export KYMA_VERSION={KYMA_RELEASE_VERSION}
+    ```
+
+## Provision a GKE or AKS cluster through Gardener
```
Comment (pbochynski) [inline docs/kyma/04-03-cluster-installation.md]:
```suggestion
## Provision a Kubernetes cluster through Gardener
```

File: docs/kyma/04-03-cluster-installation.md [hunk 3]
```diff
@@ -331,5 +318,61 @@ For MacOS, run:
...
+
+1. Create a Service Account (SA) with the required permissions in your GKE or AKS project. To learn about the SA requirements for each environment, click the question mark buttons in the **Secrets** tab of your Gardener UI.
```
Comment (pbochynski) [inline docs/kyma/04-03-cluster-installation.md]: Service Account is the term from GCP. Azure needs credentials and service principals. (Don't mention GKE and AKS in this document at all)

File: docs/kyma/04-03-cluster-installation.md [hunk 4]
```diff
@@ -331,5 +318,58 @@ For MacOS, run:
...
+
+2. Export the release version as an environment variable. Run:
+
+    ```
+    export KYMA_VERSION={KYMA_RELEASE_VERSION}
+    ```
+
+## Provision a Kubernetes cluster through Gardener
+
+1. Go to the **Secrets** tab of the Gardener UI and add secrets to enable provisioning clusters on different architectures. To learn about the requirements for each environment, click the question mark buttons.
```
Comment (klaudiagrz) [inline docs/kyma/04-03-cluster-installation.md]:
```suggestion
1. Go to the **Secrets** tab of the Gardener UI and add Secrets to enable provisioning clusters on different architectures. To learn about the requirements for each environment, click the question mark buttons.  
```

File: docs/kyma/04-03-cluster-installation.md [hunk 5]
```diff
@@ -331,5 +318,58 @@ For MacOS, run:
...
+
+    ```
+    export KYMA_VERSION={KYMA_RELEASE_VERSION}
+    ```
+
+## Provision a Kubernetes cluster through Gardener
+
+1. Go to the **Secrets** tab of the Gardener UI and add secrets to enable provisioning clusters on different architectures. To learn about the requirements for each environment, click the question mark buttons.  
+
+2. Provision a cluster form the **Clusters** tab. Choose the infrastructure you want to provision your cluster in and apply these settings:
```
Comment (klaudiagrz) [inline docs/kyma/04-03-cluster-installation.md]:
```suggestion
2. Provision a cluster form the **Clusters** tab. Choose the infrastructure in which you want to provision your cluster and apply these settings:
```

Comment (klaudiagrz) [reply docs/kyma/04-03-cluster-installation.md]: Idk, sounds better for me.

File: docs/kyma/04-03-cluster-installation.md [hunk 6]
```diff
@@ -331,5 +318,58 @@ For MacOS, run:
...
+
+  | Tab  |  Setting |  Required value |
+  |---|---|---|
+  | Infrastructure |  Kubernetes | `1.12.10`  |
+  | Worker  |  Machine type | `n1-standard-4` (GCP) <br> `Standard_D4_v3` (Azure) |
+  | Worker  | Autoscaler min.  | `3` |
+
+3. After you provision the cluster, download the kubeconfig file available under the **Show Cluster Access** option in the **Actions** column.
+
+4. Export the downloaded kubeconfig to an environment variable to connect to the cluster you provisioned. Run:
```
Comment (klaudiagrz) [inline docs/kyma/04-03-cluster-installation.md]:
```suggestion
4. Export the downloaded kubeconfig as an environment variable to connect to the cluster you provisioned. Run:
```

Comment (klaudiagrz) [reply docs/kyma/04-03-cluster-installation.md]: as / to ? 🤔

**Pull request merged**